src/erlfmt.erl

```diff
@@ -59,7 +59,10 @@ init(State) ->
 
 %% API entry point
 -spec format_file(file:name_all() | stdin, out(), config()) ->
-    {ok, [error_info()]} | skip | {error, error_info()}.
+    {ok, [error_info()]} |
+    {check_failed, binary(), binary(), [error_info()]} |
```

Comment:
Yeah, I was thinking about this one too. It seemed to be used in tests, but perhaps that's not enough to justify such API

/cc @awalterschulze

Reply:
This is very useful for testing and I was hoping this can be used to offload to a diffing function in future.
Then someone can pass a --diff flag and we only need to change erlfmt_cli.erl
But maybe this is not needed.

Another option is maybe to rather just read the file and pass it to format_string and then check if there is a diff and bypass using format_file all together

What do you think?

Reply:
If I were making decisions here, I would tell you that we can implement that when the need arises. I'm not a big fan of _futurology_ myself. But I don't have all the context that you have.
On my side, I can work with any API you like, but I do prefer the simpler ones ;)

Reply:
I have made adjustments to rather use format_string, which also simplifies the API
And started using more pattern matching instead of equals
https://github.com/WhatsApp/erlfmt/pull/133/